src/gui/src/components/explorer-grid/selected-item/tabs-overview.tsx

```diff
@@ -23,6 +36,65 @@ export const OverviewTabButton = () => {
   )
 }
 
+const useContributorAvatar = (contributor: Person) => {
+  const [avatarImg, setAvatarImg] = useState<string | null>(null)
+
+  useEffect(() => {
+    const fetchAvatar = async () => {
+      if (isRecord(contributor)) {
+        const avatar = await retrieveAvatar(contributor.email || '')
+        setAvatarImg(avatar)
+      } else {
+        const emailMatch = EMAIL_PATTERN.exec(contributor)
+        if (emailMatch) {
+          const avatar = await retrieveAvatar(emailMatch[1] || '')
+          setAvatarImg(avatar)
+        }
+      }
+    }
+
+    void fetchAvatar()
+  }, [contributor])
+
+  return avatarImg
+}
```

Comment:
The idea of having the `src/gui/src/lib/external-info.ts` helper is precisely to take all this extra logic away from the component and avoid having overly complex components with hooks everywhere.

I think the way to go here is to move this logic to `external-info.ts`, expand the returned `DetailsInfo` object to contain a `contributors` property that returns an array of type `Person & { avatar: ImageInfo}` that gets the contributor info along with the avatar info required.